pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -108,12 +109,17 @@ func NewAttachDetachController(
 		return nil, fmt.Errorf("Could not initialize volume plugins for Attach/Detach Controller: %+v", err)
 	}
 
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "attachdetach-controller"})
+
```

Comment:
Looks like there was some confusion here.

`attach_detach_controller` should accept a `recorder` as a parameter in the constructor (not constrcut a new broadcaster/recorder for itself). When `cmd/kube-controller-manager/app/controllermanager.go` calls `NewAttachDetachController` it should pass in a recorder. Which is what you were doing before. The only thing I wanted you to change was make sure that you pass the existing recorder from run method to startControllers that `controllermanager` wasn't creating two copies of broadcaster/recorder in two different places.